### ISSUE TITLE
In High contrast Aquatic and desert mode disabled control controls visually appearing as enabled controls: A11y_MonacoEditor_ No Disruption of Accessibility Features

### ISSUE THREAD

**Issue by AnKushSingh05** [microsoft/monaco-editor#4942]
### Reproducible in vscode.dev or in VS Code Desktop?

- [x] Not reproducible in [vscode.dev](https://vscode.dev) or VS Code Desktop

### Reproducible in the monaco editor playground?

- [ ] Not reproducible in [the monaco editor playground](https://microsoft.github.io/monaco-editor/playground.html)

### Monaco Editor Playground Code

```typescript

```

### Reproduction Steps

GitHubTags:
#A11yTCS; #A11ySev2; #A11yMAS; #WCAG4.3.1; #Desktopweb; #Win11; #Visual Studio Code Client; #VisualStudioCode-Web-Apr2021; #Chrome; #a11y-monacoeditor; #DevTools; #PartnerBug; # No Disruption of Accessibility Features:

Environment Details:
#URL: https://microsoft.github.io/monaco-editor/playground.html?source=v0.52.2#XQAAAAJjAQAAAAAAAABBqQkHQ5NjdMjwa-jY7SIQ9S7DNlzs5W-mwj0fe1ZCDRFc9ws9XQE0SJE1jc2VKxhaLFIw9vEWSxW3yscwzZypW_1uhs0-3YGTBOsVJ4S4FwfCDQb5-2uUHXTVAefbVSd9IzaZ_vv9VR9KugsKimARHSWlrTPOQHROITHxmAetfeRrm3VHdSe7_Etv6WpgzlH4o4eQ6dMnzwfwYcUqOiZ_ge3TaAC-S-3X8NnmdsBgoPipQD3aPDiLZZPK6l7xCFp8ZECZbOE6TnVWRQ5d9hpoWU6fPQxCmvXOHG0yQjnpBbUFcTVnmLIgKYp_MMpyC98ktGgml-3rke-U2SAGxZ_LvB8DSxLE1KkD9hE5voTWZUnA5LHB1fNAbw8DB8v_jVO8Ng
Application name: Monaco Editor
Chrome: Version 131.0.6778.205 (Official Build) (64-bit)
Windows 11 Enterprise (Version 23H2)  
OS Build 22631.3737

Repro Steps:
Hit the URL: https://microsoft.github.io/monaco-editor/playground.html?source=v0.52.2#example-creating-the-editor-hello-world.
Monaco editor page will appear. TAB to code editor and press ctrl+ F key .
Apply high contrast.
Observe that In High contrast Aquatic and desert mode disabled control controls visually appearing as enabled controls.

Actual Results:
 In High contrast Aquatic and desert mode disabled control controls visually appearing as enabled controls.

Expected Results:
The disable 'previous match and next match' button border should be visible as disable in High contrast 'aquatic and desert' modes.

User Impact:
Low vision users who depend on contrast modes will face difficulty in identifying the functionality of the control and will get confused.

Attachment:;

**Comment by meganrogge** [comment]
how are you enabling that theme?

**Comment by meganrogge** [comment]
They replied they're setting this on windows:

<img width="1315" height="497" alt="Image" src="https://github.com/user-attachments/assets/91ce5f8c-94e7-44a6-abfd-c379759c6e1f" />

**Comment by hawkticehurst** [comment]
Hi! Back with an update.

As far as I can tell / from the research I've done, Windows high contrast themes enforce an environment where there's a limited set of colors available that also must always be a 7:1 contrast ratio. The result is that any form of color adjustments we make will just straight up be overridden by Windows resulting in no power on our end to make adjustments to the color of arrows in the find/search UI.

As an example, see how color adjustments made in various places in regular light theme are ignored / overridden in high contrast theme.

Light theme (with color adjustments):

<img width="400" height="auto" alt="Image" src="https://github.com/user-attachments/assets/7dc1d5dd-2d4c-4642-ac7d-efe7bf86189b" />

Aquatic high contrast theme (adjustments are overridden):

<img width="400" height="auto" alt="Image" src="https://github.com/user-attachments/assets/251a543e-77c0-4e60-8e39-11b1da0078e8" />

I even tried to see if it was possible to adjust the arrows to be a very light gray (hoping this would be part of the limited set of colors), but it seems like it's pure white or nothing.

**Comment by pawanksing3030** [comment]
@meganrogge  we found similar control in edge browser. Press ctrl+F and will get similar control. Here we can see contrast ratio is less when no results are there and it is more when results are ther. 

<img width="2682" height="567" alt="Image" src="https://github.com/user-attachments/assets/502149b7-61bc-4ab1-b722-23b2507314e2" />
